mongo/description/server_selector.go

```diff
@@ -182,7 +182,17 @@ func (writeServerSelector) SelectServer(t Topology, candidates []Server) ([]Serv
 	case Single, LoadBalanced:
 		return candidates, nil
 	default:
-		result := []Server{}
+		selected := 0
+		for _, candidate := range candidates {
+			switch candidate.Kind {
+			case Mongos, RSPrimary, Standalone:
+				selected++
+			}
+		}
+		if selected == len(candidates) {
+			return candidates, nil
+		}
```

Comment:
```suggestion
```

We don't want to return the exact candidate slice, it could cause memory / concurrency issues. This is why we reverted https://github.com/mongodb/mongo-go-driver/commit/26e6c8ac9da113e7e4398588bd9ffa42cdc4e2ea